alpha/alpha-server/src/main/java/org/apache/servicecomb/pack/alpha/server/fsm/GrpcOmegaCallback.java

```diff
@@ -19,32 +19,71 @@
 
 import com.google.protobuf.ByteString;
 import io.grpc.stub.StreamObserver;
+import java.lang.invoke.MethodHandles;
 import org.apache.servicecomb.pack.alpha.core.OmegaCallback;
 import org.apache.servicecomb.pack.alpha.core.TxEvent;
+import org.apache.servicecomb.pack.alpha.core.exception.CompensateAskFailedException;
+import org.apache.servicecomb.pack.alpha.core.exception.CompensateConnectException;
+import org.apache.servicecomb.pack.alpha.core.fsm.CompensateAskType;
 import org.apache.servicecomb.pack.contract.grpc.GrpcCompensateCommand;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 class GrpcOmegaCallback implements OmegaCallback {
 
+  private static final Logger LOG = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
   private final StreamObserver<GrpcCompensateCommand> observer;
+  private CompensateAskWait compensateAskWait;
 
   GrpcOmegaCallback(StreamObserver<GrpcCompensateCommand> observer) {
     this.observer = observer;
   }
 
   @Override
   public void compensate(TxEvent event) {
-    GrpcCompensateCommand command = GrpcCompensateCommand.newBuilder()
-        .setGlobalTxId(event.globalTxId())
-        .setLocalTxId(event.localTxId())
-        .setParentTxId(event.parentTxId() == null ? "" : event.parentTxId())
-        .setCompensationMethod(event.compensationMethod())
-        .setPayloads(ByteString.copyFrom(event.payloads()))
-        .build();
-    observer.onNext(command);
+    compensateAskWait = new CompensateAskWait(1);
+    try {
+      GrpcCompensateCommand command = GrpcCompensateCommand.newBuilder()
+          .setGlobalTxId(event.globalTxId())
+          .setLocalTxId(event.localTxId())
+          .setParentTxId(event.parentTxId() == null ? "" : event.parentTxId())
+          .setCompensationMethod(event.compensationMethod())
+          .setPayloads(ByteString.copyFrom(event.payloads()))
+          .build();
+      observer.onNext(command);
+      compensateAskWait.await();
```

Comment:
Do we have timeout here?
I don't want to  block the compensate() calling thread.

Reply:
I plan to add compensation timeout and retry in the third step, I think to add two parameters to the @Compensable

compTimeout： compensation timeouts 
compRetries：compensation retries

Reply:
> I need to redefine the @Compensable attribute:
> 
> Rename retries to forwardRetries
> Rename timeout to forwardTimeout
> Add attribute reverseRetries
> Add attribute reverseTimeout

I have emailed this question to dev@servicecomb.apache.org
https://lists.apache.org/thread.html/a81f042ba5cc711c0df0a1fe7144b5ed6e9f20336534ccf67d77dcf1%40%3Cdev.servicecomb.apache.org%3E

Reply:
As this change will break the old code, I think it could be better if you create another PR for this part of change.

Reply:
> As this change will break the old code, I think it could be better if you create another PR for this part of change.

OK，I  have created a JIRA https://issues.apache.org/jira/browse/SCB-1646 for it